packages/mdc-toolbar/mdc-toolbar.scss

```diff
@@ -68,6 +68,7 @@ $mdc-toolbar-mobile-breakpoint: 599px;
     &--align-end {
       justify-content: flex-end;
       order: 1;
+      flex: 0;
     }
   }
```

Comment:
In order to accommodate a super long title, we also need to add:

```
.mdc-toolbar__section {
  overflow: hidden;
}

.mdc-toolbar__title {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
``` 

Here is a [pen](http://codepen.io/yeelan0319/pen/oWgNmx) for demonstrating the case.